ewaldsum_exampl requires MPI

```
[ 39%] Building CXX object core/example/longrange/CMakeFiles/Ewald.dir/ewaldsum_example.cpp.o
cd /builds/ECP-CoPA/Cabana/build/core/example/longrange && /usr/bin/c++   -I/builds/ECP-CoPA/Cabana/core/src -I/builds/ECP-CoPA/Cabana/build/core/src -isystem /root/kokkos/include  -march=native -O3 -DNDEBUG   --std=c++11 -fopenmp -o CMakeFiles/Ewald.dir/ewaldsum_example.cpp.o -c /builds/ECP-CoPA/Cabana/core/example/longrange/ewaldsum_example.cpp
core/example/longrange/CMakeFiles/Ewald.dir/build.make:62: recipe for target 'core/example/longrange/CMakeFiles/Ewald.dir/ewaldsum_example.cpp.o' failed
In file included from /builds/ECP-CoPA/Cabana/core/example/longrange/ewald.cpp:12:0,
                 from /builds/ECP-CoPA/Cabana/core/example/longrange/ewaldsum_example.cpp:13:
/builds/ECP-CoPA/Cabana/core/example/longrange/ewald.h:16:10: fatal error: mpi.h: No such file or directory
 #include "mpi.h"
          ^~~~~~~
compilation terminated.
```

--- a/core/example/longrange/CMakeLists.txt
+++ b/core/example/longrange/CMakeLists.txt
@@ -1,6 +1,6 @@
-add_executable(Direct directsum_example.cpp)
-target_link_libraries(Direct cabanacore)
 if(Cabana_ENABLE_MPI)
+  add_executable(Direct directsum_example.cpp)
+  target_link_libraries(Direct cabanacore)
   add_executable(Ewald ewaldsum_example.cpp)
   target_link_libraries(Ewald cabanacore)
 endif()

--- a/core/example/longrange/CMakeLists.txt
+++ b/core/example/longrange/CMakeLists.txt
@@ -1,7 +1,9 @@
 add_executable(Direct directsum_example.cpp)
 target_link_libraries(Direct cabanacore)
-add_executable(Ewald ewaldsum_example.cpp)
-target_link_libraries(Ewald cabanacore)
+if(Cabana_ENABLE_MPI)
+  add_executable(Ewald ewaldsum_example.cpp)
+  target_link_libraries(Ewald cabanacore)
+endif()
 
 if(Cabana_ENABLE_Cuda)
   find_package(CUDA REQUIRED)

